# Homework 0 - Gunshot Deaths in the USA

## Load dataset

In [63]:
import pandas as pd
data = pd.read_csv('gunshot_deaths.csv')
data.head()

Unnamed: 0         Intent         Gender            Age  \
0           1  None selected  None selected  None selected   
1           2  None selected  None selected  None selected   
2           3  None selected  None selected  None selected   
3           4  None selected  None selected  None selected   
4           5  None selected  None selected  None selected   

                     Race  Deaths  Population  Rate  
0           None selected   33599   316299978  10.6  
1                   White   22079   197369634  11.2  
2                   Black    7765    38896382  20.0  
3                Hispanic    3007    54049078   5.6  
4  Asian/Pacific Islander     442    16315561   2.7

In [6]:
# total number of data
len(data)

540

## Feature Analysis

In [30]:
# extract column names
cols = data.columns.tolist()[1:]
# extract group attributes -> the ones that characterize a certain type of death
group_atts = cols[0:4]
# make a dictionary out of them:
atts_dict = {group_atts[i] : i for i in range(len(group_atts))}

In [34]:
print(cols)
print(group_atts)
print(atts_dict)

['Intent', 'Gender', 'Age', 'Race', 'Deaths', 'Population', 'Rate']
['Intent', 'Gender', 'Age', 'Race']
{'Intent': 0, 'Gender': 1, 'Age': 2, 'Race': 3}


## Total number of deaths: Table vs Calculation

Total number of deaths should be the number of deaths in the row with all group attributes set to 'None selected'.
The sum of deaths in the rows with no 'None selected' attributes should add up to this value. This is going to be verified.

In [51]:
# Total number of deaths from the table:
targets = ['None selected'] * 4
total_deaths = data[(data[group_atts] == targets).all(axis=1)]['Deaths'].iloc[0]
total_deaths

33599

In [66]:
# Total number of deaths calculation:
targets = ['None selected'] * 4
sum(data[~(data[group_atts] == targets).any(axis=1)]['Deaths'])

33595

### Claim #1 -> Nearly *two-thirds* of gun deaths are *suicides*

In [81]:
targets = ['None selected'] * 4
targets[atts_dict['Intent']] = 'Suicide'
suicide_deaths = data[(data[group_atts] == targets).all(axis=1)]['Deaths'].iloc[0]
suicide_rate = suicide_deaths / total_deaths
print('Suicide deaths:', suicide_deaths)
print('Suicide rate: {:0.2f}%'.format(suicide_rate * 100))

Suicide deaths: 21058
Suicide rate: 62.67%


We see that 62.67% of all gunshot deaths are suicides, which is nearly two-thirds.

### Claim #2 -> More than *85 percent* of suicide victims are *male*

In [85]:
targets = ['None selected'] * 4
targets[atts_dict['Intent']] = 'Suicide'
targets[atts_dict['Gender']] = 'Male'
male_suicide_deaths = data[(data[group_atts] == targets).all(axis=1)]['Deaths'].iloc[0]
male_suicide_rate = male_suicide_deaths / suicide_deaths
print('Male suicide deaths:', male_suicide_deaths)
print('Male suicide rate: {:0.2f}%'.format(male_suicide_rate * 100))

Male suicide deaths: 18162
Male suicide rate: 86.25%


### Claim #3 -> Around *a third* of all gun deaths are *homicides*

In [87]:
targets = ['None selected'] * 4
targets[atts_dict['Intent']] = 'Homicide'
homicide_deaths = data[(data[group_atts] == targets).all(axis=1)]['Deaths'].iloc[0]
homicide_rate = homicide_deaths / total_deaths
print('Homicide deaths:', homicide_deaths)
print('Homicide rate: {:0.2f}%'.format(homicide_rate * 100))

Homicide deaths: 11726
Homicide rate: 34.90%


34.9% of all gun deaths are homicides, which makes up for around 1/3.

### Claim #4 -> Around *two-thirds* of homicide victims who are *males* in the *age-group of 15--34* are *black*

In [92]:
targets = ['None selected'] * 4
targets[atts_dict['Intent']] = 'Homicide'
targets[atts_dict['Gender']] = 'Male'
targets[atts_dict['Age']] = '15 - 34'
young_male_deaths = data[(data[group_atts] == targets).all(axis=1)]['Deaths'].iloc[0]

targets[atts_dict['Race']] = 'Black'
black_deaths = data[(data[group_atts] == targets).all(axis=1)]['Deaths'].iloc[0]

black_rate = black_deaths / young_male_deaths
print('Young male deaths:', young_male_deaths)
print('Young black male deaths:', black_deaths)
print('Black deaths rate: {:0.2f}%'.format(black_rate * 100))


Young male deaths: 6520
Young black male deaths: 4312
Black deaths rate: 66.13%


66.13% or around 2/3 of all young male deaths belong to the black race.

### Claim #5 -> *Women* constitute only *15 percent* of the total *homicide* victims.

In [93]:
targets = ['None selected'] * 4
targets[atts_dict['Intent']] = 'Homicide'
targets[atts_dict['Gender']] = 'Female'
women_homicide_deaths = data[(data[group_atts] == targets).all(axis=1)]['Deaths'].iloc[0]
women_homicide_rate = women_homicide_deaths / homicide_deaths
print('Women homicide deaths:', women_homicide_deaths)
print('Women homicide rate: {:0.2f}%'.format(women_homicide_rate * 100))

Women homicide deaths: 1791
Women homicide rate: 15.27%
